### PR TITLE
Segment Story Component: Removed any type from Segment story component

### PR DESCRIPTION
**Pull request by RoxanaAnamariaTurc**
**What is this feature?**
Updated the Segment Component from storybook by removing the any types.

**Why do we need this feature?**

To ensure the component is typed and less prone to errors.

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Segment/Segment.story.tsx [hunk 1]
```diff
@@ -99,7 +103,7 @@ export const GroupedArrayOptions = () => {
 };
 
 export const CustomOptionsAllowed = () => {
-  const [value, setValue] = useState(options[0]);
+  const [value, setValue] = useState<SelectableValue>(options[0]);
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Segment/Segment.story.tsx:106]:
```suggestion
  const [value, setValue] = useState<SelectableValue<string | number>>(options[0]);
```

you'll also need to remove the destructuring in the `onChange`, e.g.
```tsx
        onChange={(value) => {
          setValue(value);
          action('Segment value changed')(value);
        }}
```

File: packages/grafana-ui/src/components/Segment/Segment.story.tsx [hunk 2]
```diff
@@ -115,13 +119,13 @@ export const CustomOptionsAllowed = () => {
   );
 };
 
-const CustomLabelComponent = ({ value }: any) => <div className="gf-form-label">custom({value})</div>;
+const CustomLabelComponent = ({ value }: SelectableValue) => <div className="gf-form-label">custom({value})</div>;
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Segment/Segment.story.tsx:122]:
```suggestion
const CustomLabelComponent = ({ value }: SelectableValue<string | number>) => <div className="gf-form-label">custom({value})</div>;
```

File: packages/grafana-ui/src/components/Segment/Segment.story.tsx [hunk 3]
```diff
@@ -134,7 +138,7 @@ export const CustomLabelField = () => {
 };
 
 export const HtmlAttributes = () => {
-  const [value, setValue] = useState<any>(options[0]);
+  const [value, setValue] = useState<SelectableValue>(options[0]);
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Segment/Segment.story.tsx:141]:
```suggestion
  const [value, setValue] = useState<SelectableValue<string | number>>(groupedOptions[0].options[0]);
```

File: packages/grafana-ui/src/components/Segment/Segment.story.tsx [hunk 4]
```diff
@@ -12,14 +13,17 @@ const AddButton = (
...
+  value: value,
+});
+const options: Array<{}> = ['Option1', 'Option2', 'OptionWithLooongLabel', 'Option4'].map(toOption);
 const groupedOptions = [
   { label: 'Names', options: ['Jane', 'Tom', 'Lisa'].map(toOption) },
   { label: 'Prime', options: [2, 3, 5, 7, 11, 13].map(toOption) },
 ];
 
-const SegmentFrame = ({ options, children }: any) => (
+const SegmentFrame = ({ options, children }: { options: Object[]; children: React.ReactNode }) => (
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Segment/Segment.story.tsx:26]:
```suggestion
const SegmentFrame = ({ options, children }: { options: Array<SelectableValue<string | number>>; children: React.ReactNode }) => (
```

File: packages/grafana-ui/src/components/Segment/Segment.story.tsx [hunk 5]
```diff
@@ -50,7 +54,7 @@ const meta: Meta<typeof Segment> = {
 };
 
 export const ArrayOptionsWithPrimitiveValue = () => {
-  const [value, setValue] = useState('Option1');
+  const [value, setValue] = useState<string | number | undefined>('Option1');
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Segment/Segment.story.tsx:57]:
```suggestion
  const [value, setValue] = useState<string | undefined>('Option1');
```

File: packages/grafana-ui/src/components/Segment/Segment.story.tsx [hunk 6]
```diff
@@ -83,7 +87,7 @@ export const ArrayOptionsWithPlaceholder = () => {
 };
 
 export const GroupedArrayOptions = () => {
-  const [value, setValue] = useState<any>(groupedOptions[0].options[0]);
+  const [value, setValue] = useState<SelectableValue>(groupedOptions[0].options[0]);
   return (
     <SegmentFrame options={options}>
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Segment/Segment.story.tsx:92]:
```suggestion
    <SegmentFrame options={groupedOptions}>
```

File: packages/grafana-ui/src/components/Segment/Segment.story.tsx [hunk 7]
```diff
@@ -115,13 +119,13 @@ export const CustomOptionsAllowed = () => {
...
 };
 
-const CustomLabelComponent = ({ value }: any) => <div className="gf-form-label">custom({value})</div>;
+const CustomLabelComponent = ({ value }: SelectableValue) => <div className="gf-form-label">custom({value})</div>;
 
 export const CustomLabelField = () => {
-  const [value, setValue] = useState<any>(groupedOptions[0].options[0].value);
+  const [value, setValue] = useState<string | number | undefined>(groupedOptions[0].options[0].value);
   return (
     <SegmentFrame options={options}>
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Segment/Segment.story.tsx:127]:
```suggestion
    <SegmentFrame options={groupedOptions}>
```

File: packages/grafana-ui/src/components/Segment/Segment.story.tsx [hunk 8]
```diff
@@ -134,7 +138,7 @@ export const CustomLabelField = () => {
 };
 
 export const HtmlAttributes = () => {
-  const [value, setValue] = useState<any>(options[0]);
+  const [value, setValue] = useState<SelectableValue>(options[0]);
   return (
     <SegmentFrame options={options}>
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Segment/Segment.story.tsx:143]:
```suggestion
    <SegmentFrame options={groupedOptions}>
```

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
lots of suggestions but they basically all stem from the one change to `toOption`

shout if anything doesn't make sense 👍

File: packages/grafana-data/src/types/select.ts [hunk 9]
```diff
@@ -1,7 +1,7 @@
 /**
  * Used in select elements
  */
-export interface SelectableValue<T = any> {
+export interface SelectableValue<T> {
```
[the remark this answers is not in the document]
Comment (RoxanaAnamariaTurc) [reply packages/grafana-data/src/types/select.ts]: Thank you for the suggestions , I made the changes now :)

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

**Pull request merged**